src/settings/toml/builder.rs

```diff
@@ -0,0 +1,76 @@
+use std::env;
+use std::path::PathBuf;
+use std::process::Command;
+
+use serde::{Deserialize, Serialize};
+
+use super::ScriptFormat;
+
+const UPLOAD_DIR: &str = "dist";
+const WATCH_DIR: &str = "src";
+
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct Builder {
+    command: Option<String>,
+    #[serde(default = "project_root")]
+    pub cwd: PathBuf,
+    #[serde(default = "upload_dir")]
+    pub upload_dir: PathBuf,
+    pub upload_format: ScriptFormat,
+    #[serde(default = "watch_dir")]
+    pub watch_dir: PathBuf,
+}
+
+fn project_root() -> PathBuf {
+    env::current_dir().unwrap()
+}
+
+fn upload_dir() -> PathBuf {
+    project_root().join(UPLOAD_DIR)
+}
+
+fn watch_dir() -> PathBuf {
+    project_root().join(WATCH_DIR)
+}
+
+impl Builder {
+    pub fn verify_config(&self) -> Result<(), failure::Error> {
+        if self.upload_dir.canonicalize()? == project_root().canonicalize()?
+            || self.watch_dir.canonicalize()? == project_root().canonicalize()?
+        {
+            failure::bail!("Wrangler doesn't support using the project root as the watch directory or upload directory.");
+        }
+        if !self.upload_dir.is_dir() {
+            failure::bail!("A path was provided for upload_dir that is not a directory.");
+        }
+        if !self.watch_dir.is_dir() {
+            failure::bail!("A path was provided for watch_dir that is not a directory.");
```

Comment:
Can we mention the path that is invalid?

`{} is not a valid X_dir, since it's not a directory`

Reply:
done